plutus-use-cases/src/Language/Plutus/Coordination/Contracts/CrowdFunding.hs

```diff
@@ -0,0 +1,150 @@
+-- | Crowdfunding contract implemented using the [[Plutus]] interface.
+-- This is the fully parallel version that collects all contributions
+-- in a single transaction. This is, of course, limited by the maximum
+-- number of inputs a transaction can have.
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE RecordWildCards   #-}
+{-# OPTIONS -fplugin=Language.Plutus.CoreToPLC.Plugin #-}
+module Language.Plutus.Coordination.Contracts.CrowdFunding (
+    Campaign(..)
+    -- * Functionality for campaign contributors
+    , contribute
+    , contributionScript
+    , refund
+    , refundTrigger
+    -- * Functionality for campaign owners
+    , collect
+    , collectFundsTrigger
+    ) where
+
+import           Language.Plutus.Coordination.Plutus
+import           Language.Plutus.CoreToPLC.Plugin     (plc)
+import qualified Language.Plutus.CoreToPLC.Primitives as Prim
+
+import           Prelude                              (Bool (..), Maybe (..), Num (..), Ord (..), succ, sum, ($))
+
+-- | A crowdfunding campaign.
+data Campaign = Campaign
+    { campaignDeadline           :: !BlockHeight
+    , campaignTarget             :: !Value
+    , campaignCollectionDeadline :: !BlockHeight
+    , campaignOwner              :: !PubKey
+    }
+
+-- | Contribute funds to the campaign (contributor)
+--
+contribute :: Campaign -> Value -> TxM [TxOutRef]
+contribute c value = do
+    assert (value > 0)
+    contributorPubKey <- lookupMyPubKey
+    myPayment         <- createPayment (value + standardTxFee)
+    let validator = contributionScript c (Just contributorPubKey)
```

Comment:
I'm confused, does this mean that each contributor is going to be paying to a different script? I'd assumed they would all be paying to the same script. Maybe that's impossible, since their public key has to be involved somehow?

Reply:
Later it looks like we expect them all to be paying inputs to the same script.

Reply:
Each contributor's script contains that contributor's public key to validate a refund later on. So they all have a different hash. (It means that the campaign owner has to submit *n* different versions of the same script when they collect the funds. Not sure if there is a way around that!)

> Later it looks like we expect them all to be paying inputs to the same script.

If you are referring to the `collect` script, that's an error (just noticed that the txn model currently in Plutus doesn't let you include the actual validation script to unlock the outputs)